vpr/src/pack/re_cluster.h

```diff
@@ -15,16 +15,27 @@
 #include "cluster_util.h"
 
 /**
- * @brief This function moves an atom out of its cluster and create a new cluster for it
+ * @brief This function moves a molecule out of its cluster and create a new cluster for it
  * 
  * This function can be called from 2 spots in the vpr flow. 
  *   - First, during packing to optimize the initial clustered netlist 
  *             (during_packing variable should be true.)
  *   - Second, during placement (during_packing variable should be false)
```

Comment:
Good to briefly explain that additional state is updated in one of these cases (I think?)